nixci: 0.1.3 -> 0.2.0

srid opened the pull request:
## Description of changes

Updates nixci. Release notes: https://github.com/srid/nixci/releases/tag/0.2.0

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Commit:
nixci: 0.1.3 -> 0.2.0

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2672

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2685

pbsds commented:
Result of `nixpkgs-review pr 254996` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nixci</li>
  </ul>
</details>

---


- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64_linux
- [x] executables tested on x86_64_linux
- [x] all depending packages build
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

pbsds commented on pkgs/tools/nix/nixci/default.nix:
```diff
@@ -1,32 +1,37 @@
-  # devour-flake executable.
-  DEVOUR_FLAKE = lib.getExe devour-flake;
+  # devour-flake flake.
+  DEVOUR_FLAKE = fetchFromGitHub {
```
```suggestion
  env.DEVOUR_FLAKE = fetchFromGitHub {
```

pbsds commented on pkgs/tools/nix/nixci/default.nix:
```diff
@@ -1,32 +1,37 @@
 , openssl
 , pkg-config
 , Security
-, devour-flake
```
Should this package be removed from `pkgs/tools/nix/devour-flake/default.nix`?

srid replied:
Sure, done.

pbsds reviewed (changes requested):
builds and runs fine, but a few nits

Commit:
devour-flake: remove package

The package is no longer used by other packages. The 'nixci' package
used it until version 0.2.0.

srid commented:
Is there anything else to be done for this PR?

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1141

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1161

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -597,8 +597,6 @@ with pkgs;
 
   dec-decode = callPackage ../development/tools/dec-decode { };
 
-  devour-flake = callPackage ../tools/nix/devour-flake { };
```
Shouldn't this get an alias entry?

srid replied:
@SuperSandro2000 Note that I _removed_ (not added) `devour-flake` entirely in this PR (it is no longer necessary). What change are you proposing here exactly?

SuperSandro2000 replied:
That you add a throw to aliases.nix so that evaluation is not broken if someone for whatever reason used this.

srid replied:
Ah, understood. It is very likely that nobody has been using `pkgs.devour-flake` directly. I added it in #244487 for use in the old `nixci`.
